substrate/frame/revive/src/exec.rs

```diff
@@ -1109,7 +1129,9 @@ where
 		}
 
 		self.pop_frame(success);
-		output
+		output.map(|output| {
+			self.top_frame_mut().last_frame_output = output;
+		})
```

Comment:
maybe [inspect](https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect) instead of map here

Reply:
Inspect returns the original result; however we attach the original result to the frame instead, so we need a map.

Reply:
but you are not returning anything from map so this is pretty much equivalent to this isn' it ?

```suggestion
		output.inspect(|output| {
			self.top_frame_mut().last_frame_output = output;
		});
```

Reply:
Inspect, only provides access to a referenced value, so this won't work. I'm still not sure. `map` seems exactly what we want here?

Reply:
ah right sorry

Reply:
no worries :)